Lib/Sort.hpp

```diff
@@ -26,21 +26,44 @@
 
 namespace Lib {
 
+template<class T>
+struct HasCompareFunction 
+{ 
+  template<class U> 
+  static constexpr decltype(std::declval<U const&>().compare(std::declval<U const&>()), bool()) check(int) { return true; }
```

Comment:
Sheer sorcery!

Reply:
it is a real pain indeed. I added some documentation for usage.